internal/yamlalias/wrap.go

```diff
@@ -0,0 +1,117 @@
+// Package yamlalias adds support for aliases when parsing YAML.
+//
+// Aliases are specified as a comma-separated list specified via
+// "yaml-aliases" struct tag.
+//
+// Example:
+//
+//   type Config struct {
+//     UserName string `yaml:"username" yaml-aliases:"userName,user-name"`
+//   }
+//
+// This will parse the following YAML examples in the same way:
+//   {"username": "John Doe"}
+//   {"userName": "John Doe"}
+//   {"user-name": "John Doe"}
+//
+// Note: It only adds aliases for top-level fields in the struct.
+package yamlalias
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// Unmarshal is a wrapper for yaml.Unmarshal but adds
+// support for any yaml-aliases specified in out.
+func Unmarshal(in []byte, out interface{}) error {
+	return yaml.Unmarshal(in, addAliases(out))
+}
+
+// UnmarshalStrict is a wrapper for yaml.UnmarshalStrict but adds
+// support for any yaml-aliases specified in out.
+func UnmarshalStrict(in []byte, out interface{}) error {
+	return yaml.UnmarshalStrict(in, addAliases(out))
+}
+
+// addAliases generates a new struct that contains all exported fields
+// from the passed in struct, but adds additional fields for any
+// aliases.
+//
+// Given a struct such as:
+//
+//   type Config struct {
+//     UserName string `yaml:"username" yaml-aliases:"userName,user-name"`
```

Comment:
Worth adding what happens for non-alias cases.

```suggestion
//     UserName string `yaml:"username" yaml-aliases:"userName,user-name"`
//     TTL time.Duration `yaml:"ttl"`
```